### ISSUE TITLE
'game.end()' takes too long to quit the game on Android

### ISSUE THREAD

**Issue by mrtzd** [cocos/cocos-engine#11878]
### Cocos Creator version

3.5.2

### System information

Windows 10 Editor

### Issue description

This doesn't always happen but generally when I've played the game for a longer period and then I try to quit my game it doesn't exit immediately. Instead the game hangs for more than 1 minute and then eventually quits. I'm using the 'game.end()' function, should I do something else before I call this function? 
Also could this problem be solved by this change #10880 ?

I've tried to debug this problem by running a debug build of the game on Android device and checking the logs but it seems like this issue only occurs when using the release builds.

### Steps to reproduce

Trying to exit a game after playing for a while by calling the `game.end()` function on Android platform (only happens in release builds).

**Comment by minggo** [comment]
I think it is fixed in v3.6, but i am not sure.
@timlyeee please double check it.

**Comment by timlyeee** [comment]
Hey, @mrtzd, in v3.6 we've refactored the procedure by finishing the activity on Android, while in v3.5.2, it exits the process rudely.

But no matter for v3.6 or v3.5, they will all call dispatch close event to make sure the user's state could be saved, and tasks finished with assurance. If you've registered some js event callback on `Game.END`, they would be all called at this moment and it takes time. 

> I've tried to debug this problem by running a debug build of the game on Android device and checking the logs but it seems like this issue only occurs when using the release builds.

Could you plz show a demo that can replay this situation? It's quite weird that I haven't met it yet.

**Comment by mrtzd** [comment]
> Hey, @mrtzd, in v3.6 we've refactored the procedure by finishing the activity on Android, while in v3.5.2, it exits the process rudely.

I will definitely test this again with the new version when it comes out (I guess if I want to test it right now I have to compile the engine from 'v3.6.0' branch?)

>  If you've registered some js event callback on `Game.END`, they would be all called at this moment and it takes time.

I don't think I have registered any event callback on `Game.END` since I actually don't know how to do that.

> Could you plz show a demo that can replay this situation? It's quite weird that I haven't met it yet.

I'll try to see if I can replicate this issue with a simple demo that I can share here.

Also I tested my game on another older Android device and in this case the game quits immediately upon tapping the exit button but then shows this error message:
`
Unfortunately, [GAME_NAME] has stopped.
`
